typespec-tests/src/main/java/com/_specs_/azure/core/basic/BasicAsyncClient.java

```diff
@@ -469,6 +469,33 @@ public Mono<Response<BinaryData>> exportWithResponse(int id, String format, Requ
         return this.serviceClient.exportWithResponseAsync(id, format, requestOptions);
     }
 
+    /**
+     * Adds a user or updates a user's fields.
+     * 
+     * Creates or updates a User.
+     * 
+     * @param id The user's id.
+     * @param resource The resource instance.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return details about a user on successful completion of {@link Mono}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<User> createOrUpdate(int id, User resource) {
```

Comment:
I think we had a test calling protocol API. You can change it now to call this one.